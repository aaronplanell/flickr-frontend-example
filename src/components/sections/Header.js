import React from 'react';

const getCurrentPhotoset = (photosets, idPhotoset) => {
  if (photosets && photosets.hasOwnProperty('photoset')) {
    const listOfPhotosets = photosets.photoset.filter( (item) => {
      return item.id === idPhotoset;
    });

    if (listOfPhotosets.length === 1) {
      return listOfPhotosets[0];
    }
  }

  return null;
}

const getMessage = (photosets, idPhotoset, selectedPhoto) => {
  const currentPhotoset = getCurrentPhotoset(photosets, idPhotoset);

  // Photo selected
  if (selectedPhoto && selectedPhoto.hasOwnProperty('id'))
    return 'Click the Pockémon again to close it.';

  // Photoset selected but not photo
  if (currentPhotoset && currentPhotoset.hasOwnProperty('id'))
    return 'Click the Pockémon to see it in its original size.';

  // Neither Photo nor Photoset selected
  return 'Select a type of Pockémon: water, fire, energy,... whatever!';
}

const Header = ({photosets, idPhotoset, selectedPhoto}) => (
    <header>
      <strong>{getMessage(photosets, idPhotoset, selectedPhoto)}</strong>
    </header>
);

Header.propTypes = {
  photosets: React.PropTypes.object,
  idPhotoset: React.PropTypes.string,
  selectedPhoto: React.PropTypes.object
}

export default Header;
