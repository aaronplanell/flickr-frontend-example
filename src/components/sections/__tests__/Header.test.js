import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Header from '../Header';

const photosets = {
  photoset: [
    {
      id: 1234,
      name: 'Fire'
    }
  ]
};
const idPhotoset = '1234';
const selectedPhoto = {};

it('Header Renders OK', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Header
      photosets={photosets}
      idPhotoset={idPhotoset}
      selectedPhoto={selectedPhoto}
    />,
    div);
});

it('Header Snapshot matchs', () => {
  const tree = renderer.create(
    <Header
      photosets={photosets}
      idPhotoset={idPhotoset}
      selectedPhoto={selectedPhoto}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
