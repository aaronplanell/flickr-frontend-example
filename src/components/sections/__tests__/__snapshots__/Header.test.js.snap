exports[`test Header Snapshot matchs 1`] = `
<header>
  <strong>
    Select a type of Pock√©mon: water, fire, energy,... whatever!
  </strong>
</header>
`;
